x/concentrated-liquidity/math/math.go

```diff
@@ -173,11 +173,6 @@ func GetNextSqrtPriceFromAmount1OutRoundingDown(sqrtPriceCurrent, liquidity, amo
 // GetLiquidityFromAmounts takes the current sqrtPrice and the sqrtPrice for the upper and lower ticks as well as the amounts of asset0 and asset1
 // and returns the resulting liquidity from these inputs.
 func GetLiquidityFromAmounts(sqrtPrice, sqrtPriceA, sqrtPriceB sdk.Dec, amount0, amount1 sdk.Int) (liquidity sdk.Dec) {
-	// Reorder the prices so that sqrtPriceA is the smaller of the two.
-	if sqrtPriceA.GT(sqrtPriceB) {
-		sqrtPriceA, sqrtPriceB = sqrtPriceB, sqrtPriceA
-	}
-
```

Comment:
This needs to stay I believe. What the test was likely saying was, all the test cases were in the correct order and none of them had to be reordered

Reply:
+1

Reply:
Reasoning for the removal here was that the `Liuqiudity0` and `Liquidity1` method below ensures that regardless of sqrtPriceA and sqrtPriceB are in sequence no matter what, I can also put this check back in tho

Reply:
I think we need to be cautious removing defense in depths this late in the game, especially when we have other high pri items to focus on. Agreed on doing this in further iterations, but if we do this now, someone could not know the contract here changed and for instance, expect that this method still properly reorders when it does not.

Reply:
I have always beens confused about this flip. It feels like it turns otherwise invalid state (lower price greater than upper price) into a valid one, opening the door for lower level issues to be escalated.

If we do not have a concrete case that this is intended to handle, then I think at the very least we should error here, but I think just removing it altogether is appropriate.

Reply:
I've been confused by this check as well, I think removing it makes sense, but merits its own PR

Reply:
Decision made to revert this for now